railties/lib/rails/generators/rails/sessions/templates/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,22 @@
+class SessionsController < ApplicationController
+  allow_unauthenticated_access
+  rate_limit to: 10, within: 3.minutes, only: :create, with: -> { redirect_to new_session_url, alert: "Try again later." }
+
+  def new
+    redirect_to root_url if authenticated?
```

Comment:
I don't think this redirect will work. It looks like `Current.session` is only being set through `require_authentication` which gets skipped when using `allow_unauthenticated_access`. So `Current.session` would always end up nil here.

Reply:
Fixed in https://github.com/rails/rails/commit/1dd37e01937f6711cd15267481df8c0446961472